### PR TITLE
[Storage] Encryption Scope SAS

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
TODO record unit tests.  Waiting on Encryption Scope SAS and the 2020-10-02 service version to be deployed to canary,

File: sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs [hunk 1]
```diff
@@ -606,6 +606,105 @@ public async Task AppendBlockAsync_EncryptionScope()
             Assert.AreEqual(TestConfigDefault.EncryptionScope, response.Value.EncryptionScope);
         }
 
+        [RecordedTest]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/20501")]
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:610]: can this be playbackonly?

File: sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace Azure.Storage.Test.Shared
         StorageVersionExtensions.LatestVersion,
         StorageVersionExtensions.MaxVersion,
         RecordingServiceVersion = StorageVersionExtensions.MaxVersion,
-        LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion })]
+        LiveServiceVersions = new object[] { StorageVersionExtensions.MaxVersion })]
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs:34]: why this change?

File: sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:610]: Nope, because I am not able to record this test.  Identity SAS only works in canary and prod, and this feature and the 2020-10-02 service version are not currently enabled in canary.

File: sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs:34]: TODO undo

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs [hunk 3]
```diff
@@ -85,6 +86,7 @@ public static TestConstants Create<TStorageTestEnvironment>(StorageTestBase<TSto
                     ContentEncoding = test.GetNewString(),
                     ContentLanguage = test.GetNewString(),
                     ContentType = test.GetNewString(),
+                    EncryptionScope = test.GetNewString(),
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs:89]: here

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs:89]: How is this breaking all those tests?
